blockchain_storage/src/store.rs

```diff
@@ -436,26 +459,24 @@ mod test {
     #[test]
     fn test_storage_transactions_with_result_check() {
         common::concurrency::model(|| {
-            use storage::DbTransaction;
-
             // Set up the store and initialize the version to 2
             let mut store = Store::new_empty().unwrap();
             assert_eq!(store.set_storage_version(2), Ok(()));
 
             // Concurrently bump version by 3 and 5 in two separate threads
             let thr0 = {
-                let mut store = store.clone();
+                let store = Store::clone(&store);
                 common::thread::spawn(move || {
-                    let mut tx = store.start_transaction();
+                    let mut tx = store.start_transaction_rw();
                     let v = tx.get_storage_version().unwrap();
                     assert!(tx.set_storage_version(v + 3).is_ok());
                     assert!(tx.commit().is_ok());
                 })
             };
             let thr1 = {
-                let mut store = store.clone();
+                let store = Store::clone(&store);
                 common::thread::spawn(move || {
-                    let mut tx = store.start_transaction();
+                    let mut tx = store.start_transaction_rw();
```

Comment:
No tests for read-only?

Reply:
Will revisit this once again and potentially add more tests or make some read-only if possible.